pkg/auth/dependency/webapp/template.go

```diff
@@ -304,14 +304,14 @@ var TemplateAuthUILoginHTML = template.Spec{
 				</div>
 				{{ end }}{{ end }}
 
-				{{ if .x_login_id_input_type }}{{ if and (not (eq .x_login_id_input_type "phone")) .x_login_id_input_type_has_text }}
-				<input class="input text-input primary-txt" type="text" name="x_login_id" placeholder="{{ localize "login-id-placeholder" }}" value="{{ .x_login_id }}">
+				{{ if .x_login_id_input_type }}{{ if (not (eq .x_login_id_input_type "phone")) }}
+				<input class="input text-input primary-txt" type="{{ .x_login_page_login_id_input_type }}" name="x_login_id" placeholder="{{ localize "login-id-placeholder" }}" value="{{ .x_login_id }}">
 				{{ end }}{{ end }}
 
-				{{ if .x_login_id_input_type }}{{ if and (eq .x_login_id_input_type "phone") .x_login_id_input_type_has_text }}
-				<a class="link align-self-flex-start" href="{{ call .MakeURLWithQuery "x_login_id_input_type" "text" }}">{{ localize "use-text-login-id-description" }}</a>
+				{{ if .x_login_id_input_type }}{{ if eq .x_login_id_input_type "phone" }}
+				<a class="link align-self-flex-start" href="{{ call .MakeURLWithQuery "x_login_id_input_type" .x_login_page_login_id_input_type }}">{{ localize "use-text-login-id-description" }}</a>
 				{{ end }}{{ end }}
-				{{ if .x_login_id_input_type }}{{ if and (not (eq .x_login_id_input_type "phone")) .x_login_id_input_type_has_phone }}
+				{{ if .x_login_id_input_type }}{{ if (not (eq .x_login_id_input_type "phone")) }}
```

Comment:
After updating, the link `Use phone number instead` is shown if I enable `username` and `email` only.